hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/selective/SelectiveWorkBalancer.java

```diff
@@ -57,7 +58,7 @@ private int countMissingResources(SubscriptionAssignmentView state) {
     }
 
     private void equalizeWorkload(SubscriptionAssignmentView state, SubscriptionAssignmentView.Transformer transformer) {
-        if (state.getSubscriptionsCount() > 1) {
+        if (state.getSubscriptionsCount() > 1 && !state.getConsumerNodes().isEmpty()) {
```

Comment:
this should protect us agains different weird effects, because there are methods inside that do not tolerate empty set of consumer nodes